coalib/parsing/LineParser.py

```diff
@@ -30,7 +30,9 @@ def __init__(self,
                                             "section" despite of the current
                                             section.
         """
-        section_name_surroundings = section_name_surroundings or {"[": "]"}
+        section_name_surroundings = (
+                {"[": "]"} if section_name_surroundings is None
+                else section_name_surroundings)
```

Comment:
I am sorry I got a bit busy the past two days.... I will make the changes today and push them. I have never used `MappingProxyType`. So, please review it too after my push.